src/components/UI/JSONSchemaForm/ArrayFieldTemplate/index.tsx

```diff
@@ -1,39 +1,95 @@
 import { ArrayFieldTemplateProps } from '@rjsf/utils';
-import { Box } from 'grommet';
-import React from 'react';
+import { Accordion, AccordionPanel, Box, Text } from 'grommet';
+import React, { useState } from 'react';
+import styled from 'styled-components';
 import Button from 'UI/Controls/Button';
-import InputGroup from 'UI/Inputs/InputGroup';
+
+const StyledButton = styled(Button)`
+  height: 40px;
+`;
+
+const Icon = styled(Text)<{ isActive?: boolean }>`
+  transform: rotate(${({ isActive }) => (isActive ? '0deg' : '-90deg')});
+  transform-origin: center center;
+  transition: 0.15s ease-out;
+`;
 
 const ArrayFieldTemplate: React.FC<ArrayFieldTemplateProps> = ({
   items,
   canAdd,
   title,
+  schema,
   onAddClick,
 }) => {
+  const [activeIndexes, setActiveIndexes] = useState<number[]>([]);
+
+  const itemsAreObjects =
+    schema.items &&
+    typeof schema.items !== 'boolean' &&
+    !Array.isArray(schema.items) &&
+    schema.items.type === 'object';
```

Comment:
In the future I would like to find a better way to determine if the array items are objects or primitives (string, boolean, etc.)